File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        hashmap = {}

        for num in nums:
            hashmap[num] = 1 + hashmap.get(num,0)
        
        arr = []
        for key in hashmap:
            arr.append([hashmap[key], key])
        
        arr.sort();

        sol = []

        for i in range(k):
            sol.append(arr.pop()[1])
        
        return sol

File: Data Structures & Algorithms/remove-element/submission-2.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int swapIndex = nums.length - 1;
        int runnerIndex = 0;
        
        while(runnerIndex <= swapIndex){
            if (nums[runnerIndex] != val){
                runnerIndex += 1;
            } else {
                nums[runnerIndex] = nums[swapIndex];
                nums[swapIndex] = val;
                swapIndex--;
            }
        }

        return runnerIndex;
    }
}